haystack/document_stores/weaviate.py

```diff
@@ -887,36 +889,100 @@ def query(
                             }
                             ```
         :param top_k: How many documents to return per query.
+        :param all_terms_must_match: Whether all terms of the query must match the document.
```

Comment:
I'd just do: 
```
:param all_terms_must_match: Not used in weaviate
```

Reply:
Agreed. I am changing this as suggested and sending in the new commit.